### PR TITLE
Fix ratelimit key

### PR DESCRIPTION
**Pull request by jenshu**
BOT NOTES: 
resolves https://github.com/solo-io/skv2/issues/404

### CODE REVIEW

File: contrib/pkg/input/input_reconciler.go [hunk 1]
```diff
@@ -110,31 +111,42 @@ func (r *inputReconciler) reconcileEventsForever(reconcileInterval time.Duration
...
 	}
-	defer r.queue.Done(key)
+	defer r.queue.Done(queueItem)
 
-	var isRemoteCluster bool
+	// get the string representation of the queue item
+	key, ok := queueItem.(string)
+	if !ok {
+		contextutils.LoggerFrom(r.ctx).Errorw("got a work queue item of non-string type", "item", queueItem)
+		r.queue.Forget(queueItem)
```
Comment (nfuden) [inline contrib/pkg/input/input_reconciler.go:124]: should we return after the forgets?

Comment (conradhanson) [reply contrib/pkg/input/input_reconciler.go:124]: new line 159 is showing that this is the default behavior so I'd think we would want to

Comment (jenshu) [reply contrib/pkg/input/input_reconciler.go:124]: ha yeah good catch, both places where we error, we should return

**Review by conradhanson — CHANGES_REQUESTED** [review]
LGTM just needs a changelog

**Review by nfuden — APPROVED** [review]

**Review by conradhanson — APPROVED** [review]
expects on exact error strings in tests are usually brittle (should prob compare by error type instead), but since this code shouldn't be changing much I think it's fine.

**Review by nfuden — APPROVED** [review]

**Pull request merged**